test/integration/stats_integration_test.cc

```diff
@@ -275,5 +300,33 @@ TEST_P(ClusterMemoryTestRunner, MemoryLargeClusterSizeWithRealSymbolTable) {
   EXPECT_MEMORY_LE(m_per_cluster, 36000);
 }
 
+TEST_P(ClusterMemoryTestRunner, MemoryLargeHostSizeWithStats) {
+  // A unique instance of ClusterMemoryTest allows for multiple runs of Envoy with
+  // differing configuration. This is necessary for measuring the memory consumption
+  // between the different instances within the same test.
+  const size_t m1000 = ClusterMemoryTestHelper::computeMemoryDelta(1, 1, 1, 1001, true);
+  const size_t m_per_host = (m1000) / 1000;
+
+  // Note: if you are increasing this golden value because you are adding a
+  // stat, please confirm that this will be generally useful to most Envoy
+  // users. Otherwise you are adding to the per-host memory overhead, which
+  // will be significant for Envoy installations configured to talk to large
+  // numbers of hosts.
+  //
+  // History of golden values:
+  //
+  // Date        PR       Bytes Per Host      Notes
+  //                      exact upper-bound
+  // ----------  -----    -----------------   -----
+  // 2019/09/09  8189     2883         3100   Initial per-host memory snapshot
+
+  // Note: when adjusting this value: EXPECT_MEMORY_EQ is active only in CI
+  // 'release' builds, where we control the platform and tool-chain. So you
+  // will need to find the correct value only after failing CI and looking
+  // at the logs.
+  EXPECT_MEMORY_EQ(m_per_host, 2883);
+  EXPECT_MEMORY_LE(m_per_host, 3100);
```

Comment:
thanks for doing this!  Can you also make a RealSymbolTable variant, with
```
  Stats::TestUtil::SymbolTableCreatorTestPeer::setUseFakeSymbolTables(false);
```
as the first line, and updated target memory values?

FYI if you didn't already know this you can test this locally with:
```
 --compilation_mode=opt --test_env=ENVOY_MEMORY_TEST_EXACT=true
```

Reply:
Memory usage without fake symbol tables is 2739

I rather keep a single variant of this test since it takes a while to run and upcoming changes in the branch below remove use of symbol tables for host stats.  How about I have the test disable fake symbol tables?

https://github.com/envoyproxy/envoy/compare/master...antoniovicente:stats_primitive_host_stats

Reply:
sounds good.

Reply:
Done.  Waiting for CI to process the change.